ozaria/engine/cinematic/commands/CinematicParser.js

```diff
@@ -62,15 +62,17 @@ const parseSetup = (shot, systems) =>
 /**
  * Returns an array of commands.
  * This is required as we update this list to add commands to the start and end.
- * @param {DialogNode} dialogNode
- * @param {Object} systems
+ * @param {Object} arg
+ * @param {DialogNode} arg.dialogNode
+ * @param {Object} arg.systems
+ * @param {Object} arg.shot the current shot
  * @returns {AbstractCommand[]}
  */
-const parseDialogNode = (dialogNode, systems) => {
+const parseDialogNode = ({ dialogNode, systems, shot }) => {
   const dialogCommands = Object.values(systems)
     .filter(sys => sys !== undefined && typeof sys.parseDialogNode === 'function')
     .reduce((commands, sys) => {
-      const dialogCommands = sys.parseDialogNode(dialogNode)
+      const dialogCommands = sys.parseDialogNode(dialogNode, shot)
```

Comment:
This function has now been extended to optionally pass in the shot. This allows systems to look at the broader scope of data in their current shot.